tests/MongoDB.Driver.Core.Tests/Core/Operations/AsyncCursorTests.cs

```diff
@@ -34,16 +34,161 @@
 using MongoDB.Driver.Core.TestHelpers;
 using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
 using MongoDB.Driver.Core.WireProtocol;
+using MongoDB.Driver.Core.WireProtocol.Messages;
 using MongoDB.Driver.Core.WireProtocol.Messages.Encoders;
 using Moq;
 using Xunit;
 
 namespace MongoDB.Driver.Core.Operations
 {
-    // public methods
     public class AsyncCursorTests
     {
         // public methods
+        [Fact]
+        public void Close_and_dispose_should_dispose_cursor_only_once()
+        {
+            int testCursorId = 1;
+            var mockChannelHandle = new Mock<IChannelHandle>();
+            var mockChannelSource = new Mock<IChannelSource>();
+            SetupChannelMocks(mockChannelSource, mockChannelHandle, false, $"{{ 'ok' : true, 'cursorsNotFound' : [], 'cursorsKilled' : [{testCursorId}] }}");
+
+            var subject = CreateSubject(cursorId: 1, channelSource: Optional.Create(mockChannelSource.Object));
+            subject.Close();
+            mockChannelSource.Verify(c => c.GetChannel(It.IsAny<CancellationToken>()), Times.Once);
+            subject.Dispose();
+            mockChannelSource.Verify(c => c.GetChannel(It.IsAny<CancellationToken>()), Times.Once);
```

Comment:
What is this Verify actually proving?

Reply:
It checks that KillCursorCommand (both versions) was called as it's the only one operation which requires channel in Close/Dispose methods, I didn't find another way how to check that Kill operation was launched.